source/server/config/http/zipkin_http_tracer.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/server/instance.h"
+
+#include "server/configuration_impl.h"
+
+namespace Envoy {
+namespace Server {
+namespace Configuration {
+
+/**
+ * Config registration for the zipkin http tracer. @see HttpTracerFactory.
+ */
+class ZipkinHttpTracerFactory : public HttpTracerFactory {
+public:
+  // HttpTracerFactory
+  Tracing::HttpTracerPtr tryCreateHttpTracer(const std::string& type,
```

Comment:
This has the added benefit of you can fail if you try to register something with the same name. So would be great to add test for that also, and fix all the networking filters too. Bigger change, but I think well worth it.